nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,298 @@ def save(self, *args, **kwargs):
+                        raise ValidationError("Value must be a datetime or str object.")
+                else:
+                    # if value is a datetime object
+                    # check if datetime object has tzinfo
+                    if value.tzinfo is None:
+                        value = value.replace(tzinfo=timezone.utc)
+                    value = value.replace(microsecond=0).isoformat()
+            # Validate selected choice
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_SELECT:
+                if value not in self.metadata_type.choices.values_list("value", flat=True):
+                    raise ValidationError(
+                        f"Invalid choice ({value}). Available choices are: {', '.join(self.metadata_type.choices.values_list('value', flat=True))}"
+                    )
+
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_MULTISELECT:
+                if isinstance(value, str):
+                    value = value.split(",")
+                if not set(value).issubset(self.metadata_type.choices.values_list("value", flat=True)):
+                    raise ValidationError(
+                        f"Invalid choice(s) ({value}). Available choices are: {', '.join(self.metadata_type.choices.values_list('value', flat=True))}"
+                    )
+            self._value = value
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+            if self.metadata_type != database_object.metadata_type:
+                raise ValidationError({"metadata_type": "Cannot be changed once created"})
+
+        # Check if there is any intersections of scoped_fields of ObjectMetadata instances in the database.
+        object_metadata_scoped_fields = (
+            ObjectMetadata.objects.filter(
+                metadata_type=self.metadata_type,
+                assigned_object_type=self.assigned_object_type,
+                assigned_object_id=self.assigned_object_id,
+            )
+            .exclude(pk=self.pk)
+            .values_list("scoped_fields", flat=True)
+        )
+        duplicate_scoped_fields_list = set([])
+
+        # TODO Make sure that scoped fields are all direct fields on the assigned_object_type model
+
+        for scoped_fields in object_metadata_scoped_fields:
+            duplicate_scoped_fields_list |= set(self.scoped_fields).intersection(scoped_fields)
+            # if self.scoped_fields or scoped_fields are [] which means all fields are scoped
+            # That means any fields in self.scoped_fields or scoped_fields are considered overlapped
+            if not self.scoped_fields or not scoped_fields:
+                raise ValidationError(
+                    {
+                        "scoped_fields": f"There are other Object Metadata instances of metadata type {self.metadata_type} scoping all the fields for {self.assigned_object}"
```

Comment:
Feel like this is not the clearest error message for the case where `self.scoped_fields` is the one covering all fields?

Reply:
Separated the two conditions and gave them distinct error messages.